packages/react-components/src/menu/src/Menu.tsx

```diff
@@ -20,55 +24,64 @@ import {
 import { MenuContext } from "./MenuContext";
 import { MenuItem } from "./MenuItem";
 import { MenuSection } from "./MenuSection";
-import { NodeType, useCollection } from "../../collection";
-import { SelectionMode } from "./selectionMode";
-import { any, arrayOf, bool, elementType, func, number, oneOf, oneOfType, string } from "prop-types";
-import { forwardRef } from "react";
 import { isNil, isNumber } from "lodash";
+import type { SelectionMode } from "./selectionMode";
 
 export const ItemKeyProp = "data-o-ui-key";
 
-const propTypes = {
+export interface InnerMenuProps extends DOMProps {
     /**
      * A controlled set of the selected item keys.
      */
-    selectedKeys: arrayOf(string),
+    selectedKeys?: string[]
     /**
      * The initial value of `selectedKeys` when uncontrolled.
      */
-    defaultSelectedKeys: arrayOf(string),
+    defaultSelectedKeys?: string[]
     /**
      * Called when the selected keys change.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @param {String[]} keys - The keys of the selected items..
      * @returns {void}
      */
-    onSelectionChange: func,
+    onSelectionChange?(event: SyntheticEvent, keys: string[]): void
     /**
      * The type of selection that is allowed.
      */
-    selectionMode: oneOf(["none", "single", "multiple"]),
+    selectionMode?: SelectionMode;
     /**
      * Whether or not the menu should autofocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number;
     /**
      * Default focus target when enabling autofocus.
      */
-    defaultFocusTarget: string,
+    defaultFocusTarget?: string;
     /**
      * Whether or not the listbox take up the width of its container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
      * React children.
      */
-    children: oneOfType([any, func])
-};
+    children: ReactNode;
+    /**
+     * @ignore
+     */
+    "aria-label"?: string;
```

Comment:
Maybe it's in another PR but I wonder why a `DomProps` interface has been introduce but not an `AriaLabelingProps ` interface? IMO it's the coolest one ;)

Reply:
other PR